Rest.Services/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,37 @@
+using Rest.Models;
+namespace Rest.Services;
+public class ProductRepository : IRepository<long, Product>
+{
+    public Dictionary<long, Product> _productDictionary;
+    public long _id;
+
+    public ProductRepository()
+    {
+            _productDictionary = new Dictionary<long, Product>();
+            _id = 0;
+    }
+
+    public void Add(Product entity)
+    {
+        _id ++;
+        _productDictionary.Add(_id, entity);
```

Comment:
I don't think that using a Dictionary here is a good idea. I suggest to use an `IEnumerable` or a `List`. That will eliminate the necessity to handle an incremental `id`.

Another thing is, you don't want to handle `id's` order manually.

Reply:
I think adding the `id` is not a bad approach, because if he deletes an element from the list, then ids will change and you wouldn't like that to happen.

For instance:

[1,2,3,4] and you delete 2, then you could potentially have duplicated ids next time you add a new element.

Reply:
@Aryevang as @reynoldmorel said this approach helps me to not duplicate IDs, because I always increment this attribute in each ```Add()``` method. Otherwise, using a ```List``` as example I could add a ```Status``` property to ```Product Class``` to change it from "active" to "inactive" when I use ```Remove()``` method, instead of removing the object. It could be a future refactor I guess.

Reply:
Got it.